CalendarApp/Notification Handler/NotificationHandler.swift

```diff
@@ -0,0 +1,160 @@
+    }
+    
+    func registerNotifications(with completion: @escaping() -> Void) {
+        center.requestAuthorization(options: [.alert, .badge, .sound]) { (granted, error) in
+            completion()
+        }
+    }
+    
+    func deleteReminderWithID(_ reminderID: UUID) {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "reminderID == %@",
+                                        reminderID as CVarArg)
+        do {
+            let notifications = try context.fetch(request)
+            if notifications.count == 1 {
+                context.delete(notifications[0])
+                guard let reminderID = notifications[0].reminderID else {
+                    return
+                }
+                center.removePendingNotificationRequests(withIdentifiers: [reminderID.uuidString])
+                try context.save()
+            }
+        } catch {
+            // TODO: error handling
+        }
+    }
+    
+    func fetchReminderInfo() -> [Reminder] {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "archived == false")
+        do {
+            let reminders = try context.fetch(request)
+            return reminders
+        } catch {
+            return [Reminder]()
+        }
+    }
+    
+    func updateReminderForEvent(_ event: Event, _ date: Date) {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "eventID == %@",
+                                        event.objectUUID as CVarArg)
+        do {
+            let notifications = try context.fetch(request)
+            if notifications.count == 1 {
+                context.delete(notifications[0])
+                guard let reminderID = notifications[0].reminderID else {
+                    return
+                }
+                center.removePendingNotificationRequests(withIdentifiers: [reminderID.uuidString])
+                scheduleNotification(event: event, date: date)
+                try context.save()
+            }
+        } catch {
+            // TODO: error handling
+        }
+    }
+    
+    func deleteReminderForEvent(_ eventID: UUID) {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "eventID == %@",
+                                        eventID as CVarArg)
+        do {
+            let notifications = try context.fetch(request)
+            if notifications.count == 1 {
+                context.delete(notifications[0])
+                guard let reminderID = notifications[0].reminderID else {
+                    return
+                }
+                center.removePendingNotificationRequests(withIdentifiers: [reminderID.uuidString])
+                try context.save()
+            }
+        } catch {
+            // TODO: error handling
+        }
+    }
+    
+    func checkReminderForEvent(_ eventID: UUID) -> Date? {
```

Comment:
This is a little hard to reason about. Check can mean a lot of things. This does check if a reminder exists for an event ID but it also returns the date. Maybe `dateOfReminder(eventIdentifier: UUID)`?